src/main/java/StringManipulation.java

```diff
@@ -1,32 +1,107 @@
+/*
+    Jon Kaimmer
+    CS410 Software Engineering
+    Spring 2023
+    Junit Testing Assignment
+
+    Spring Manipulation
+ */
+
+import java.lang.IllegalArgumentException;
+import java.lang.IndexOutOfBoundsException;
+import java.lang.NullPointerException;
+import java.util.*;
+
+
 public class StringManipulation implements StringManipulationInterface {
 
+    public String s;
+
     @Override
     public String getString() {
-        return null;
+        return s;
     }
 
     @Override
     public void setString(String string) {
```

Comment:
In the event that user sets string = "", consider overriding that input with null? May be outside of how Sara expects us to implement, though perhaps worth considering for methods below. Save on calling isEmpty(). For sure up for interpretation, no single answer.